include/fmt/base.h

```diff
@@ -8,9 +8,9 @@
 #ifndef FMT_BASE_H_
 #define FMT_BASE_H_
 
-#include <limits.h>  // CHAR_BIT
-#include <stdio.h>   // FILE
-#include <string.h>  // strlen
+#include <climits>  // CHAR_BIT
+#include <cstdio>   // FILE
+#include <cstring>  // strlen
```

Comment:
We intentionally use `.h` headers here since C++ wrappers add some build time overhead (small but nonzero) and `fmt/base.h` is optimized for build speed.

Reply:
Thanks for the clarification. I changed them because of the following warning:

```console
fmt/include/fmt/base.h:12:10: warning: '#include <filename>' attaches the declarations to the named module 'fmt', which is not usually intended; consider moving that directive before the module declaration [-Winclude-angled-in-module-purview]
C/C++:    12 | #include <stdio.h>   // FILE
C/C++:       |          ^
C/C++: 1 warning generated.
```

So If I enable `-Werror` globally, it won't compile. Any suggestion about it?

Reply:
We can make these conditionally included in non-modular build (i.e. when `FMT_EXPORT` is not defined).

Reply:
Okay, I will also add a compile defition `FMT_MODULE` for this prupose.

Reply:
Sounds good.

Reply:
Done